### PR TITLE
Lab 5 MouseEvent

### PR DESCRIPTION
**Pull request by Lola9614**

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (kowallus) [inline .idea/compiler.xml]: project files to be removed!

File: src/main/java/edu/kis/powp/jobs2d/features/DrawerFeature.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 	public static void setupDrawerPlugin(Application application) {
 		SelectClearPanelOptionListener selectClearPanelOptionListener = new SelectClearPanelOptionListener();
 
+
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawerFeature.java]: unnecessary change

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 3]
```diff
@@ -95,6 +98,47 @@ private static void setupLogger(Application application) {
 		application.addComponentMenuElement(Logger.class, "Severe level",
 				(ActionEvent e) -> logger.setLevel(Level.SEVERE));
 		application.addComponentMenuElement(Logger.class, "OFF logging", (ActionEvent e) -> logger.setLevel(Level.OFF));
+
+
+		application.getFreePanel().addMouseListener(new MouseListener(){
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: extract to src/main (separate class and configuration routine)

**Review by kowallus — CHANGES_REQUESTED** [review]

File: .idea/compiler.xml (hunk 1 above)
Comment (Lola9614) [reply .idea/compiler.xml]: resolved

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java (hunk 3 above)
Comment (Lola9614) [reply src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: resolved

File: src/main/java/edu/kis/powp/jobs2d/features/DrawerFeature.java (hunk 2 above)
Comment (Lola9614) [reply src/main/java/edu/kis/powp/jobs2d/features/DrawerFeature.java]: resolved

File: src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.appbase.Application;
+import javax.swing.*;
+import java.awt.event.MouseListener;
+import java.util.logging.Logger;
+
+public class MouseEvents implements MouseListener {
+
+    private Application application;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java]: Inject JPanel instead of Application

File: src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    }
+
+    @Override
+    public void mouseReleased(java.awt.event.MouseEvent mouseEvent) {
+        int widthOfPanel=application.getFreePanel().getWidth()/2;
+        int heightOfPanel = application.getFreePanel().getHeight()/2;
+
+        if (SwingUtilities.isLeftMouseButton(mouseEvent)) {
+            DriverFeature.getDriverManager().getCurrentDriver().operateTo(mouseEvent.getX() - widthOfPanel,
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java]: Inject DriverManager in constructor

**Review by kowallus — CHANGES_REQUESTED** [review]

File: src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java (hunk 5 above)
Comment (Lola9614) [reply src/main/java/edu/kis/powp/jobs2d/events/MouseEvents.java]: resolved
